src/main/java/com/microsoft/sqlserver/jdbc/IOBuffer.java

```diff
@@ -3312,67 +3313,60 @@ void writeDouble(double value) throws SQLServerException {
      *            the source JDBCType
      * @param precision
      *            the precision of the data value
+     * @param scale
+     *            the scale of the column
+     * @throws SQLServerException
      */
     void writeBigDecimal(BigDecimal bigDecimalVal,
             int srcJdbcType,
-            int precision) throws SQLServerException {
+            int precision,
+            int scale) throws SQLServerException {
         /*
          * Length including sign byte One 1-byte unsigned integer that represents the sign of the decimal value (0 => Negative, 1 => positive) One 4-,
          * 8-, 12-, or 16-byte signed integer that represents the decimal value multiplied by 10^scale. The maximum size of this integer is determined
          * based on p as follows: 4 bytes if 1 <= p <= 9. 8 bytes if 10 <= p <= 19. 12 bytes if 20 <= p <= 28. 16 bytes if 29 <= p <= 38.
          */
 
-        boolean isNegative = (bigDecimalVal.signum() < 0);
-        BigInteger bi = bigDecimalVal.unscaledValue();
-        if (isNegative)
-            bi = bi.negate();
+        /*
+         * setScale of all BigDecimal value based on metadata as scale is not sent seperately for individual value. set RoundingMode to avoid lose of
+         * precision. Say, for BigDecimal("0.1"), if scale in metdadata is 0, then ArithmeticException would be thrown if RoundingMode is not set
+         */
+        try {
+            bigDecimalVal = bigDecimalVal.setScale(scale, RoundingMode.HALF_UP);
+        }
+        catch (ArithmeticException e) {
+            MessageFormat form = new MessageFormat(SQLServerException.getErrString("R_valueOutOfRange"));
+            Object[] msgArgs = {SSType.DECIMAL};
+            throw new SQLServerException(form.format(msgArgs), SQLState.DATA_EXCEPTION_DATETIME_FIELD_OVERFLOW, DriverError.NOT_SET, e);
+        }
```

Comment:
Either use `RoundingMode.ROUND_UNNECESSARY` or remove the `try/catch` as it will never be thrown with `RoundingMode.HALF_UP`.   My preference is to use `ROUND_UNNECESSARY` as it forces the user to decide how to deal with loss of precision rather than having it silently lose precision.

Also the exception message has a SQL State of `DATA_EXCEPTION_DATETIME_FIELD_OVERFLOW`.  Is there a `DATA_EXCEPTION_DECIMAL_FIELD_OVERFLOW` equivalent that can be used instead?